### ISSUE TITLE
cmd+k to clear terminal is surprisingly slow

### ISSUE THREAD

**Issue by marwan-at-work** [zed-industries/zed#10176]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

To reproduce:

1. Open the Zed terminal. 
2. Press "enter" a few times to create multi line
3. Type "cmd+k" to clear the screen. 
4. Note how it takes a few seconds to do it. 
5. Compare with typing "clear" instead of cmd+k to notice the difference in speed. 

### Environment

Zed: v0.129.2 (Zed)
OS: macOS 14.3.0
Memory: 96 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

https://github.com/zed-industries/zed/assets/16294261/4cfe9644-5e82-40a9-b2a3-59b28983190a

**Comment by showme032** [comment]
Yeah I have the same issue, even though in one post I saw it was apparently fixed. Cmd+K still takes about a second after typing to clear the console.

**Comment by notpeter** [comment]
I can no longer reproduce this and I think this was likely fixed.

@marwan-at-work can you confirm this is not longer an issue for you?

**Comment by mikayla-maki** [comment]
This was an issue with our action dispatch and was fixed a little while ago.

**Comment by showme032** [comment]
I still have it. Although it seems to have somewhat improved, there is still a noticeable delay between command input and action

**Comment by marwan-at-work** [comment]
Yes there is still a pretty noticeable delay running `Zed: v0.139.3`

**Comment by mikayla-maki** [comment]
Ahh, I thought I remembered a fix going out.

**Comment by GraphtyLove** [comment]
Same issue with `Zed 0.139.3`. Running on a Macbook Air M2.

**Comment by marwan-at-work** [comment]
If it helps w/ debugging: if you press almost any key after cmd+k, then the screen is cleared quickly without any delay.

**Comment by stalkerg** [comment]
Okey, it's not only for the terminal! I am using ctrl-k to delete the lines, and I have the same delay and the same behavior (any new input is a clear view). 
If I set this binding to any other configuration, it does not happen. Something else is triggered for ctrl/cmd-k. 
I built Zed from the source a few moments ago, the latest git.

**Comment by GraphtyLove** [comment]
It looks like a vim binding. 
When you press `CMD+k` it looks like Zed is waiting for another key to be pressed to do something.
Like a macro record or something.

When you press any key after that, the symbol in bottom disappear and the terminal is cleared.

![image](https://github.com/user-attachments/assets/117d0ad7-1a07-4e72-b6de-f01c778e6d7d)

**Comment by notpeter** [comment]
I wanted to suggest a potential workaround until this is fixed: `ctrl-l`.

This will clear the terminal without delay.  Note it serves a different function.  `cmd-k` clears the screen AND the scrollback buffer, while `ctrl-l` just clear the screen and leaves the scrollback buffer alone.

**Comment by stalkerg** [comment]
for me it's became working fine, is it still an issue for you?

**Comment by marwan-at-work** [comment]
@GraphtyLove hi there, I just tested it and it's still slow for me. 

```
$ zed --version
Zed 0.155.2 – /Applications/Zed.app
```

Let me know if there's anything to do or to help with on my side. Thanks!

**Comment by dangh** [comment]
@marwan-at-work I think you need to add `cmd-k` key to your keymap to disable the default behavior. This is my keymap:

```json
{
  "context": "Terminal",
  "bindings": {
    "cmd-k": "terminal::Clear"
  }
}
```

**Comment by marwan-at-work** [comment]
@dangh thank you that fixed it! However, it should not be necessary for everyone to create the above keymap config, that should be the default behavior unless I'm missing something.

**Comment by mikayla-maki** [comment]
I believe this has been fixed now, I can't reproduce the problem in a standard zed installation with no custom keymaps, with or without vim mode enabled.

Note to people who find this issue in the future: if there is a keybinding collision (e.g. binding both `cmd-k` and `cmd-k left`), Zed will wait to see if you type the following key for one second, then dispatch the action. Check if you have a keymap that is colliding with a zed keybinding.

**Comment by injust** [comment]
Also, this interacts very poorly with the new `which_key` feature.

I have `which_key` configured with a 500 ms delay. When I press `cmd-k` in the terminal, Zed waits 500 ms, shows me the `which_key` panel, then waits a bit more and decides I want to clear the screen, so the `which_key` panel then closes itself.

**Comment by notpeter** [comment]
Emacs keymap on windows/linux is also super broken because of similar Ctrl-k conflicts.

**Comment by injust** [comment]
> This issue is NOT fixed, when i clear it, it still takes a few seconds to clear. I am on the newest version of Zed.

But it can be worked around. You likely have a chorded user keybind that includes cmd-k, for which you should change the context to include `&& !Terminal`.
